replica/replica.go

```diff
@@ -1115,10 +1157,13 @@ func (r *Replica) unmarshalFile(file string, obj interface{}) error {
 	if err != nil {
 		return err
 	}
-	defer f.Close()
 
-	dec := json.NewDecoder(f)
-	return dec.Decode(obj)
+	err = json.NewDecoder(f).Decode(obj)
+	if err != nil {
+		return err
```

Comment:
f.close is missing here

Reply:
why this change required?

Reply:
We were ignoring error thrown by f.Close().